packages/patternfly-4/react-core/src/components/Checkbox/Checkbox.js

```diff
@@ -1,24 +1,62 @@
 import React from 'react';
 import styles from '@patternfly/patternfly-next/components/Check/check.css';
-import { css } from '@patternfly/react-styles';
 import PropTypes from 'prop-types';
+import { css, getModifier } from '@patternfly/react-styles';
 
 const propTypes = {
-  /** additional classes added to the Checkbox */
+  /** Additional classes added to the Checkbox. */
   className: PropTypes.string,
   /** Flag to show if the Checkbox selection is valid or invalid. */
   isValid: PropTypes.bool,
   /** Flag to show if the Checkbox is disabled. */
   isDisabled: PropTypes.bool,
   /** A callback for when the Checkbox selection changes. */
-  onChange: PropTypes.func
+  onChange: PropTypes.func,
+  /** Label text of the checkbox. */
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
+  /** Id of the checkbox. */
+  id: props => {
+    if (!props.id) {
+      return new Error('CheckBox must have an id.');
+    }
+    if (typeof props.id !== 'string') {
+      return new Error(`Id must be string. Received ${typeof props.id}.`);
+    }
+    return null;
+  },
+  /** Aria-label of the checkbox. */
+  'aria-label': props => {
```

Comment:
If using something like react-intl or react-i18next this will likely always error. I think we should just check for the existence of it since we are allowing nodes in addition to strings.